http/src/main/scala/org/http4s/blaze/http/http2/client/ClientStage.scala

```diff
@@ -0,0 +1,198 @@
+  }(Execution.trampoline)
+
+  private[this] class BodyReaderImpl extends BodyReader {
+    // We don't want to call `release()` here because we may be waiting for this message
+    // to be written, so we don't want to close the stream
+    override def discard(): Unit = observeEOF()
+
+    override def isExhausted: Boolean = inboundConsumed
+
+    override def apply(): Future[ByteBuffer] = {
+      if (inboundConsumed) BufferTools.emptyFutureBuffer
+      else channelRead().map {
+        case d@DataFrame(eos, data) =>
+          if (eos) discard()
+          logger.debug(s"Received data frame: $d")
+          data
+
+        case other =>
+          logger.debug(s"Received frame other than data: $other. Discarding remainder of body.")
+          discard()
+          BufferTools.emptyBuffer
+      }(Execution.directec)
+    }
+  }
+
+  private def collectResponseFromHeaders(body: BodyReader, hs: Seq[(String, String)]): Try[ReleaseableResponse] = {
+    logger.debug(s"Received response headers: $hs")
+
+    val regularHeaders = new VectorBuilder[(String, String)]
+    var pseudos = true
+    var statusCode = -1
+
+    val it = hs.iterator
+    while (it.hasNext) {
+      val pair@(k, v) = it.next()
+
+      if (!k.startsWith(":")) {
+          pseudos = false // definitely not in pseudos anymore
+          regularHeaders += pair
+      } else if (!pseudos) {
+        return Failure(new Exception("Pseudo headers were not contiguous"))
+      } else k match {
+          // Matching on pseudo headers now
+        case StageTools.Status =>
+          if (statusCode != -1)
+            return Failure(new Exception("Multiple status code HTTP2 pseudo headers detected in response"))
+
+          try statusCode = v.toInt
+          catch { case ex: NumberFormatException => return Failure(ex) }
+
+        case _ => // don't care about other pseudo headers at this time
+      }
+    }
+
+    if (statusCode != -1) Success(new ReleasableResponseImpl(statusCode, "UNKNOWN", regularHeaders.result(), body))
+    else {
+      val ex = Http2Exception.PROTOCOL_ERROR.rst(-1, "HTTP2 Response headers didn't include a status code.")
+      release(Command.Error(ex))
+      Failure(ex)
+    }
+  }
+
+  private[this] def shutdownWithError(ex: Throwable, phase: String): Unit = {
+    logger.debug(ex)(s"$name shutting down due to error in phase $phase")
+    if (_result.tryFailure(ex)) {
+      // Since the user won't be getting a `ReleasableResponse`, it is our job
+      // to close down the stream.
+      val command = if (ex == Command.EOF) Disconnect else Command.Error(ex)
+      release(command)
+    }
+  }
+}
+
+private object ClientStage {
+  private[client] def makeHeaders(request: HttpRequest): Try[Vector[(String, String)]] = {
+    UrlComposition(request.url).map { breakdown =>
+      val hs = new VectorBuilder[(String, String)]
+
+      // h2 pseudo headers
+      hs += StageTools.Method -> request.method.toUpperCase
```

Comment:
I guess we've ascertained that method doesn't include funny characters, but I still twitch when I see this without a `Locale.ROOT`.

Reply:
We haven't checked the characters, so I'll add a sane `Locale`.

Reply:
`Local.ENGLISH` it is. Thanks for the keen eye.

Reply:
Somewhere along the way, I picked up `Locale.ROOT` as "the neutral locale", but I can't articulate any difference between the two.  The important thing here is that this isn't broken in Turkey.